src/CBT.Core/Internal/ModulePropertyGenerator.cs

```diff
@@ -107,17 +108,26 @@ public bool Generate(string outputPath, string extensionsPath, string[] beforeMo
             return true;
         }
 
-        private void AddImports(ProjectRootElement project, IEnumerable<string> modulePaths)
+        private void AddImports(ProjectRootElement project, IEnumerable<PackageIdentityWithPath> modulePaths)
         {
-            foreach (ProjectImportElement import in modulePaths.Where(i => !String.IsNullOrWhiteSpace(i)).Select(modulePath => project.AddImport(Path.Combine(modulePath, ImportRelativePath))))
+            foreach (var modulePath in modulePaths.Where(i => !string.IsNullOrWhiteSpace(i?.Path)))
             {
-                import.Condition = $" Exists('{import.Project}') ";
+                bool v1Package = Directory.Exists(Path.Combine(modulePath.FullPath, ModuleRelativePathV1));
+                string importPath = v1Package ? Path.Combine(ModuleRelativePathV1, "$(MSBuildThisFile)") : Path.Combine(ImportRelativePath, Path.GetFileName(project.FullPath));

```

Comment:
Won't this expect a `build.props` instead of `Package.Id\build\Package.Id.props`?